Fix #20264: Create stress_test_acceptance_test.yml

jayam04 opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes #20264
3. This PR does the following: 
    - Create a new stress test file that runs acceptance tests few times.
    - Extracts acceptance test job into a common action file.


## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [ ] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [x] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [x] I have written tests for my code.
- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [ ] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).



<!--
If this PR affects production server data, please follow
[these instructions](https://github.com/oppia/oppia/wiki/Testing-jobs-and-other-features-on-production#submitting-a-pr-with-a-new-job-or-feature-that-requires-third-party-api)
and link to the job request doc here.

Otherwise, please delete this section.
-->

- [ ] A testing doc has been written: ... (ADD LINK) ...


- [ ] _(To be confirmed by the server admin)_ All jobs in this PR have been verified on a live server, and the PR is safe to merge.

## Proof that changes are correct
1. https://github.com/jayam04/oppia/actions/runs/18269983472/job/52011113699

    When test fails in acceptance tests:
    <img width="773" height="253" alt="image" src="https://github.com/user-attachments/assets/3c1c8cf0-5b7f-41a5-9c0e-a15dc070a52d" />

2. **When the acceptance test fails.**
    Failed test: https://github.com/oppia/oppia/actions/runs/19468597019/job/55711756217?pr=23535
    Logs uploaded: [test-logs_interested-donor_learn-about-the-organization_mobile_original.zip](https://github.com/user-attachments/files/23619850/test-logs_interested-donor_learn-about-the-organization_mobile_original.zip)
    Failure point screenshots uploaded: [failure-point-snapshots_interested-donor_learn-about-the-organization_mobile_original.zip](https://github.com/user-attachments/files/23619857/failure-point-snapshots_interested-donor_learn-about-the-organization_mobile_original.zip)

3. **When Invalid Suite Name is passed.**
    Job URL: https://github.com/jayam04/oppia/actions/runs/19692026309/job/56411637982?pr=95
    <img width="962" height="425" alt="image" src="https://github.com/user-attachments/assets/a9daded0-5a52-411c-9a46-cb7d7b299e28" />

4. When test fails, video recording is uploaded.
    <img width="971" height="358" alt="image" src="https://github.com/user-attachments/assets/efceb55b-9995-45e3-836b-746e2eef296c" />
    Uploaded Video: https://github.com/user-attachments/assets/6f2062ec-b967-444d-b3d8-9cd345f14a30



<!--
Add before-and-after videos/screenshots of the user-facing interface (including
the browser devtools console) to demonstrate that the changes made in this PR
work correctly. Make sure the actions taken in the before and after videos are
the same. (For changes involving responsiveness or adjustment of UI elements,
this should be a video that gradually resizes the viewport from wide to narrow
and back again.) If this PR is for a developer-facing feature, provide
videos/screenshots of the developer-facing interface instead.

When you make updates to the PR, please update these videos/screenshots as well.
You can drop videos/screenshots from previous versions of the PR.

The above should be done for all PRs, including short ones (e.g. a single-line change).
However, if the changes in your PRs are autogenerated via a script and you cannot
provide proof for the changes then please leave a comment "No proof of changes
needed because {{Reason}}" and remove all the sections below.
-->



<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->



<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->



<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

jayam04 committed:
Create stress_test_acceptance_test.yml

jayam04 committed:
Update stress_test_acceptance_test.yml

jayam04 committed:
Update stress_test_acceptance_test.yml

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,143 @@
+        default: '10'
+        type: string
+      test_suite:
+        description: 'Acceptance test to run.'
```
In the description, could you give an example of a valid value for this field?

jayam04 replied:
Done.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,143 @@
+    steps:
+      - name: Check for broken suites
+        id: check_skip
+        # TODO(#22448): Remove once all suites are fixed.
```
Why do we need to skip this, since the stress test isn't meant to block PRs? If a developer enters a broken suite then shouldn't this still run (and fail)?

jayam04 replied:
Yes, we don't need this, removed.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,143 @@
+name: Stress Test (Acceptance Test)
```
Maybe you can add some comments to the top of this workflow file explaining that it's not meant to be a PR check, but can be run in order to identify flaky tests (basically, explain how to use it).

jayam04 replied:
Done.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,143 @@
+        with:
+          name: desktop-acceptance-test-failure-screenshot-${{steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME}}
+          path: /home/runner/work/oppia/oppia_full_stack_test_failure_screenshots/acceptance
+        # TODO(#22667): We are running this step twice, as snapshot gets deleted if 2nd attempt passes.
```
Isn't this old code? I thought we dropped it in the other PR you merged.

It would probably be better if these steps were abstracted into a common action so that they can be used in this workflow and the current acceptance test workflow, without duplicating the code. If not feasible to do now, maybe file an issue for it?

jayam04 replied:
Updated with common action.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,143 @@
+          name: png-snapshots-${{steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME}}-new
+          path: core/tests/puppeteer-acceptance-tests/specs/**/*.png
+
+  # New job to collect all PNG snapshots into a single artifact
```
Could you explain why doing this is necessary?

jayam04 replied:
I added this to get latest PNG for screenshot comparasion in tests. However, now we are using a common action, I don't think we can get this.

Removed.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,143 @@
+      - name: Upload Desktop test recordings as Artifacts
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v4
+        with:
```
When uploading these and other artifacts, do we need to include the number of the run?

jayam04 replied:
We need to use test run, else the artifact names will be same and create a conflict while uploading them.

seanlip reviewed (changes requested):
Thanks @jayam04! Left some initial feedback, PTAL. 

In https://github.com/jayam04/oppia/actions/runs/18269983472/job/52011113699, why do the steps start with \[D\] and \[M\]? Maybe you can make that a bit clearer, e.g. [Desk], [Mob] (and drop the backslashes).

Also in that same link, why do we upload snapshots if there are no errors? https://github.com/jayam04/oppia/actions/runs/18269983472 And if each snapshot is 46 MB how come all combined snapshots is also 46 MB? I wonder if we can save space here since the main point of this script is to detect whether a suite is flaky (so let's have its output focus on that -- I think the main thing needed is to have full output in the error case).

seanlip commented:
@jayam04 Please assign @U8NWXD for subsequent reviews as I will be travelling this week. Thanks!

jayam04 committed:
Extract acceptance tests steps into common action file.

jayam04 committed:
Revert It: Updates a test to fail for checking purposes.

jayam04 committed:
Remove timout-minutes from action file

jayam04 committed:
fix: missing shell option in action file

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/logged-in-learner/add-and-remove-exploration-from-play-later.spec.ts:
```diff
@@ -65,7 +65,7 @@ describe('Logged-in User', function () {
     await loggedInUser.addLessonToPlayLater('Negative Numbers', true);
     await loggedInUser.expectToastMessage(
-      "Successfully added to your 'Play Later' list."
+      "Successfully wasn't added to your 'Play Later' list."
```
This test is failing in the CI. Also, are you sure if this is the correct message? Because it doesn't sound right.

jayam04 replied:
@imchristie I have updated it to verify that test fails properly, I will revert it.

imchristie reviewed (commented):
Completed a full review on one acceptance tests file.

jayam04 committed:
Some fixes

jayam04 committed:
Addresses comments.

U8NWXD commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,77 @@
+# This workflow is used to run the acceptance tests multiple times.
+# It's not used in the CI, but is used to test the falkiness of an acceptance test.
+
+name: Stress Test (Acceptance Test)
+
+permissions: read-all
+on:
+  workflow_dispatch:
+    inputs:
+      num_runs:
+        description: 'Number of times to run the acceptance tests (Max: 128).'
+        required: true
+        default: '10'
+        type: string
+      test_suite:
+        description: 'Acceptance test to run. (e.g. "interested-donor/make-a-donation")'
+        required: true
+        type: string
```
Please validate these inputs to make sure we block any code injection attacks. As a second line of protection, please also use GitHub's [recommended mitigation strategies](https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks) (intermediate environment variables are probably your best bet) when using these user-provided, and therefore potentially malicious, inputs

jayam04 replied:
Done.

jayam04 committed:
Sanitize Inputs

jayam04 committed:
Removed extra new-line at the end of file.

jayam04 commented:
@U8NWXD Addressed [your comment](https://github.com/oppia/oppia/pull/23535#discussion_r2485113013). PTAL!

U8NWXD commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+      - name: Sanitize test_suite input
+        run: |
+          if [[ "$TEST_SUITE" =~ [^a-zA-Z/-] ]]; then
+          echo "TEST_SUITE contains invalid characters (only alphabets, slashes, and dashes allowed)."
```
```suggestion
          echo "TEST_SUITE contains invalid characters (only letters, slashes, and dashes allowed)."
```

U8NWXD commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+    steps:
+      - name: Sanitize test_suite input
+        run: |
+          if [[ "$TEST_SUITE" =~ [^a-zA-Z/-] ]]; then
+          echo "TEST_SUITE contains invalid characters (only alphabets, slashes, and dashes allowed)."
+          exit 1
+          else
+          echo "TEST_SUITE is valid."
+          fi
```
please indent the if and else blocks for readability

jayam04 replied:
Done.

U8NWXD commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+        id: generate_suite_name_for_artifacts
+        # Replace slashes in the filename with underscores.
+        run: |
+          SUITE_NAME=${{ inputs.test_suite }}-${{ matrix.run }}-${{ matrix.mobile && 'mobile' || 'desktop' }}
```
please use the environment variable here instead of `${{ inputs.test_suite }}` to prevent code injection

jayam04 replied:
Done.

jayam04 committed:
Update .github/workflows/stress_test_acceptance_test.yml

Co-authored-by: U8N WXD <U8NWXD@users.noreply.github.com>

jayam04 committed:
Address comments.

jayam04 commented:
@U8NWXD Addressed your comments.

seanlip commented on .github/actions/run-acceptance-tests/action.yml:
```diff
@@ -0,0 +1,83 @@
+name: Run Acceptance Test
+description: Runs the acceptance tests using python installation.
```
I think you can just say "Runs a specific acceptance test", or similar -- I don't think you need to be explicit about "using python installation".

Though I am confused because the filename is run-acceptance-tests and the name here is "run acceptance test". Is this supposed to run one test or multiple tests? Please clarify and ensure that the names/descriptions/filenames are all consistent.

jayam04 replied:
Done.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+# This workflow is used to run the acceptance tests multiple times.
+# It's not used in the CI, but is used to test the falkiness of an acceptance test.
```
falkiness --> flakiness

jayam04 replied:
Done.

seanlip commented on .github/actions/run-acceptance-tests/action.yml:
```diff
@@ -0,0 +1,83 @@
+
+inputs:
+  test-suite:
+    description: Acceptance test suite to run.
```
Acceptance --> The acceptance

Give an example of a valid input as well.

jayam04 replied:
Done.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -0,0 +1,83 @@
+  test-suite:
+    description: Acceptance test suite to run.
+    required: true
+  mobile:
```
Can you call this "viewport-type", and allow it to only have one of two possibilities: "mobile" or "desktop"?

That makes the two choices more "equal". The description should be "The viewport type to use for running the tests (either 'mobile' or 'desktop')." There should be some validation that the input is one of those two choices.

jayam04 replied:
Done.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -0,0 +1,83 @@
+    description: Whether to run the tests on mobile or desktop.
+    required: true
+    type: boolean
+  screen-recording:
```
enable-screen-recording

jayam04 replied:
Done.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -0,0 +1,83 @@
+    description: Whether to record the screen during the test.
+    required: true
+    type: boolean
+  modified-suite-name:
```
modified-suite-name-for-uploads

jayam04 replied:
Done.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -0,0 +1,83 @@
+    required: true
+    type: boolean
+  modified-suite-name:
+    description: A unique suite name without spaces to use in the artifacts uploads. If not provided, the test data will not be uploaded.
```
in --> for

Can you add validation to reject the input fails to match a particular charset/regex? E.g. you want to prohibit stuff like slashes, double-dots, etc. as well -- so easiest is to just specify the regex of allowed characters.

jayam04 replied:
Done. Changed in to for and also added a step "Check if modified-suite-name-for-uploads contains invalid characters" to validate it.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -0,0 +1,83 @@
+      id: run-acceptance-test
+      shell: bash
+      # We use xvfb-run to create a virtual screen to run tests without headless mode.
+      # Numbers in 1920x1080x24 represents screen width, height and color depth respectively.
```
represents --> represent

jayam04 replied:
Done.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -0,0 +1,83 @@
+      uses: actions/upload-artifact@v4
+      with:
+        name: acceptance-test-logs-${{ inputs.modified-suite-name }}
+        path: acceptance_test_${{ inputs.modified-suite-name }}.log
```
Let's take the opportunity to clean up some of the naming. Could we follow the following conventions for the artifacts that show up in the summary page (note the difference between hyphens and underscores):

- diff-snapshots_{{modified-suite-name}}
- screen-captures_{{modified-suite-name}}
- test-logs_{{modified-suite-name}}
- video-recordings_{{modified-suite-name}}

What do you think? (Do we actually need the acceptance-test at the start?)

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -0,0 +1,83 @@
+        name: acceptance-test-logs-${{ inputs.modified-suite-name }}
+        path: acceptance_test_${{ inputs.modified-suite-name }}.log
+
+    - name: Upload test failure screenshots
```
Why do we mention "failure" here and not in the other uploads, despite them all running on failure?

jayam04 replied:
It's because these screenshots are taken when test fails. However, logs, screen-recordings, etc are for whole test and not just failure point.

seanlip replied:
Ah ok. Let's call this "Upload screenshots taken on test failure" then.

jayam04 replied:
Done.

seanlip commented on .github/workflows/full_stack_tests.yml:
```diff
@@ -208,81 +208,33 @@ jobs:
+          test-suite: ${{ matrix.suite.name }}
+          mobile: false
+          screen-recording: true
+          modified-suite-name: desktop-${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}
```
${{...}}_desktop_original (change hyphen to underscore, move to the end)

jayam04 replied:
Done.

seanlip commented on .github/workflows/full_stack_tests.yml:
```diff
@@ -292,86 +244,25 @@ jobs:
+          test-suite: ${{ matrix.suite.name }}
+          mobile: true
+          screen-recording: true
+          modified-suite-name: mobile-${{ steps.generate_suite_name_for_artifacts.outputs.MODIFIED_SUITE_NAME }}
```
${{...}}_mobile_original (change hyphen to underscore, move to the end)

(The "original" is in case we want to add a "rerun" in the future.)

jayam04 replied:
Done.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+        description: 'Number of times to run the acceptance tests (Max: 128).'
+        required: true
+        default: '10'
+        type: string
```
Shouldn't this be a number? Otherwise someone could enter "abc".

jayam04 replied:
Numbers are not supported. Ref: https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/. So, as @U8NWXD suggested, we validate it before generating matrix from the number.

seanlip replied:
Oh I see, thanks. Can I suggest using a dropdown then, have it be 1, 2, 3, 5, 10, 20, 50, 100? I don't see why, in practice, we'd need the granularity of being able to pick every integer.

jayam04 replied:
Done.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+        default: '10'
+        type: string
+      test_suite:
+        description: 'Acceptance test to run. (e.g. "interested-donor/make-a-donation")'
```
Name of acceptance test suite to run (e.g. "interested-donor/make-a-donation")

jayam04 replied:
Done.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+  workflow_dispatch:
+    inputs:
+      num_runs:
+        description: 'Number of times to run the acceptance tests (Max: 128).'
```
Add comment (with docs link) to explain where 128 comes from

jayam04 replied:
Done.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+      - name: Generate build files
+        uses: ./.github/actions/generate-build-files
+
+  acceptance_test_with_python_installation:
```
Do you need the 'python installation' bit? I think that's going to become our only installation.

jayam04 replied:
Done.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -0,0 +1,99 @@
+      fail-fast: false
+      matrix:
+        run: ${{ fromJson(needs.generate-matrix.outputs.matrix) }}
+        mobile: [false, true]
```
viewport-type: ['desktop', 'mobile']

jayam04 replied:
Done.

seanlip reviewed (changes requested):
Thanks @jayam04, took a pass.

Also @jayam04 could you please address my questions [here](https://github.com/oppia/oppia/pull/23535#pullrequestreview-3303871329)? Namely:

- In https://github.com/jayam04/oppia/actions/runs/18269983472/job/52011113699, why do the steps start with [D] and [M]? Maybe you can make that a bit clearer, e.g. [Desk], [Mob] (and drop the backslashes).

- Also in that same link, why do we upload snapshots if there are no errors? https://github.com/jayam04/oppia/actions/runs/18269983472 And if each snapshot is 46 MB how come all combined snapshots is also 46 MB? I wonder if we can save space here since the main point of this script is to detect whether a suite is flaky (so let's have its output focus on that -- I think the main thing needed is to have full output in the error case).

jayam04 committed:
Address few comments.

jayam04 committed:
Addressed few more comments.

jayam04 commented:
> Thanks @jayam04, took a pass.
> 
> Also @jayam04 could you please address my questions [here](https://github.com/oppia/oppia/pull/23535#pullrequestreview-3303871329)? Namely:
>
>     * In https://github.com/jayam04/oppia/actions/runs/18269983472/job/52011113699, why do the steps start with [D] and [M]? Maybe you can make that a bit clearer, e.g. [Desk], [Mob] (and drop the backslashes).

@seanlip I used `\` to escape `[` however it didn't worked properly. It's fixed now. I have also updated it to use **Desk** and **Mob** instead of just their initials.

>     * Also in that same link, why do we upload snapshots if there are no errors? https://github.com/jayam04/oppia/actions/runs/18269983472 And if each snapshot is 46 MB how come all combined snapshots is also 46 MB? I wonder if we can save space here since the main point of this script is to detect whether a suite is flaky (so let's have its output focus on that -- I think the main thing needed is to have full output in the error case).

I was uploading all snapshots in each-case. This was primarily used to capture snapshots for new tests in GSoC project so I didn't had to run each test manually in different environment and viewport. It's not required in this testing so I have removed this.

jayam04 committed:
Address remaining comments.

jayam04 commented:
@seanlip Addressed your comments. PTAL!

jayam04 committed:
Removed duplicate action files

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -1,72 +1,81 @@
-name: Run Acceptance Test
-description: Runs the acceptance tests using python installation.
+name: Run a specific acceptance test
+description: Runs a specific acceptance test-suite that is given as an input.
```
"test-suite" --> "test suite"

jayam04 replied:
Done.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -1,72 +1,81 @@
 inputs:
   test-suite:
-    description: Acceptance test suite to run.
+    description: 'The acceptance test suite to run(e.g. "interested-donor/make-a-donation").'
```
Add space before "("

jayam04 replied:
Done.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -1,72 +1,81 @@
     required: true
-  mobile:
+  viewport-type:
     description: Whether to run the tests on mobile or desktop.
```
The description should be "The viewport type to use for running the tests."

jayam04 replied:
Done. Previously, I got duplicate actions file and thus some changes were missed.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -1,72 +1,81 @@
+    - name: Check if modified-suite-name-for-uploads contains invalid characters.
+      run: |
+        if [[ ! "${{ inputs.modified-suite-name-for-uploads }}" =~ ^[a-zA-Z0-9_-]+$ ]]; then
+          echo "Modified suite name for uploads contains one of the invalid characters ('/', '\', '_')"
```
This doesn't match the regex. You'll want to say something like "The modified suite name should contain only the following characters: letters, digits, underscore and hyphen."

jayam04 replied:
Done.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -1,72 +1,81 @@
+      run: |
+        if [[ ! "${{ inputs.modified-suite-name-for-uploads }}" =~ ^[a-zA-Z0-9_-]+$ ]]; then
+          echo "Modified suite name for uploads contains one of the invalid characters ('/', '\', '_')"
+          exit 1
```
Just to check, if this exits, then does the run stop here or are the "if (always() && x = failure)" steps below run as well?

jayam04 replied:
I need to check this and will add to proof of changes. You can keep it open until then.

Also, I think we should switch to `failure()` instead of `always()`. Because, I have seen that when we cancel workflow manually, then the steps with `always()` may run if the condition is satisfied.

seanlip replied:
Hm, it's a bit tricky because if e.g. an upload action fails you still want those actions to run. So the always() has a purpose, I think...

There may not be a perfect solution here. Maybe evaluate some options and pick the least bad one?

jayam04 replied:
@seanlip I have changed it to `!cancelled()`. Also, I have added proof of changes showing that below steps run too.

seanlip commented on .github/actions/run-a-specific-acceptance-test/action.yml:
```diff
@@ -0,0 +1,92 @@
+      if: ${{ always() && steps.run-acceptance-test.outcome == 'failure' && inputs.modified-suite-name-for-uploads != '' }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: acceptance-test-failure-screenshot-${{ inputs.modified-suite-name-for-uploads }}
```
I think [this comment](https://github.com/oppia/oppia/pull/23535#discussion_r2514093950) might have got missed, so I'll repeat it again:

----

Let's take the opportunity to clean up some of the naming. Could we follow the following conventions for the artifacts that show up in the summary page (note the difference between hyphens and underscores):

- diff-snapshots_{{modified-suite-name}}
- failure-point-snapshots_{{modified-suite-name}}
- test-logs_{{modified-suite-name}}
- video-recordings_{{modified-suite-name}}

What do you think? (Do we actually need the acceptance-test at the start?)

Also for snapshots, note that "diff-snapshots" is for the screenshot testing and "failure-point-snapshots" is for the screenshots taken on-failure at the end.

jayam04 replied:
Done.

seanlip reviewed (changes requested):
Thanks @jayam04, reviewed, PTAL!

Also please note most of the CI tests are failing (I think you have an error in one of the scripts).

jayam04 committed:
Address comments.

jayam04 committed:
Revert "Revert It: Updates a test to fail for checking purposes."

This reverts commit b5e79419b981c8b139d3797304a755772545a8c1.

jayam04 commented:
@U8NWXD Friendly ping.

seanlip commented on .github/workflows/stress_test_acceptance_test.yml:
```diff
@@ -14,7 +14,15 @@ on:
+        type: choice
+        options:
+          - '1'
+          - '2'
```
You missed '3'. Not that important either way, but I thought it might be better to give folks some flexibility at the lower end.

jayam04 replied:
I thought you gave an example and didn't included '3'. I have added it now.

seanlip reviewed (commented):
@jayam04 it looks great! Just the leftover comment and one more small thing, and it's good to go :)

jayam04 committed:
Addressed one comment

jayam04 committed:
Fix: changed screen-recording to enable-screen-recording in action calls

jayam04 committed:
Updates always() to !cancelled()

jayam04 committed:
Merge branch 'develop' into stress-test-acceptance

jayam04 commented:
@seanlip I have addressed your comments. I still need to add proof of changes for the following:
* Invalid modified suite name passed.
* Video being uploaded as artifacts.
* Action file works properly in stress test.

However, I am struggling running the stress test in my fork. The workflow is failing in Merge Develop and Setup Dependencies. Once I find a fix, I will add above proof too. Hopefully, but EOD today. 

Let me know if anything is still remaining.

seanlip reviewed (approved):
Thanks @jayam04, seems reasonable. Please provide proof that it works and then feel free to submit!

seanlip commented:
@jayam04 Also, really great job on this :) I like how the workflow is now quite streamlined and we're reusing the logic. Thank you for all the cleanups, I think this will be a useful tool for the team!

jayam04 committed:
Updates stress_test_acceptance_test to use valid arg.

jayam04 committed:
Fix test recording not being uploaded.
